src/composables/addressBook.ts

```diff
@@ -64,19 +64,23 @@ export const useAddressBook = createCustomScopedComposable(() => {
     ])(Object.values(addressBook.value)),
   );
 
+  const addressBookFilteredByProtocol = computed(
+    () => filterAddressBook(Object.values(addressBook.value)),
+  );
+
   function setProtocolFilter(filter: Protocol | null) {
     showBookmarked.value = false;
     protocolFilter.value = filter;
   }
 
-  function setShowBookmarked(value: boolean) {
-    protocolFilter.value = null;
+  function setShowBookmarked(value: boolean, resetProtocolFilter = false) {
+    protocolFilter.value = resetProtocolFilter ? null : protocolFilter.value;
     showBookmarked.value = value;
   }
 
-  function clearFilters() {
+  function clearFilters(resetProtocolFilter = false) {
     showBookmarked.value = false;
-    protocolFilter.value = null;
+    protocolFilter.value = resetProtocolFilter ? null : protocolFilter.value;
```

Comment:
I suggest to avoid reassigning the same value because in most cases this forces the layour to re-render. Consider this:
```
if (resetProtocolFilter) {
  protocolFilter.value = null;
}
```